pandas/core/tools/datetimes.py

```diff
@@ -14,12 +15,24 @@
 from pandas.core.dtypes.common import (
     ensure_object, is_datetime64_dtype, is_datetime64_ns_dtype,
     is_datetime64tz_dtype, is_float, is_integer, is_integer_dtype,
-    is_list_like, is_numeric_dtype, is_object_dtype, is_scalar)
-from pandas.core.dtypes.generic import ABCDataFrame, ABCIndexClass, ABCSeries
+    is_list_like, is_numeric_dtype, is_scalar)
+from pandas.core.dtypes.generic import (
+    ABCDataFrame, ABCDatetimeIndex, ABCIndex, ABCIndexClass, ABCSeries)
 from pandas.core.dtypes.missing import notna
 
+from pandas._typing import ArrayLike, DatetimeScalar
 from pandas.core import algorithms
 
+# ---------------------------------------------------------------------
+# types used in annotations
+
+
+DatetimeScalarOrArrayConvertible = Union[DatetimeScalar, list, tuple,
```

Comment:
define the types here (that you used in _typing). The reason is that these are *convertible* to datetimes, not datetimes themselves. The convertible is a much wider range of types.

Reply:
defined